File: backend/src/Spisa.Domain/Entities/Invoice.cs
using Spisa.Domain.Common;

namespace Spisa.Domain.Entities;

public class Invoice : BaseEntity
{
    public string InvoiceNumber { get; set; } = string.Empty;
    public long SalesOrderId { get; set; }
    public DateTime InvoiceDate { get; set; }
    public decimal NetAmount { get; set; }
    public decimal TaxAmount { get; set; }
    public decimal TotalAmount { get; set; }
    public decimal? UsdExchangeRate { get; set; }
    public bool IsPrinted { get; set; } = false;
    public DateTime? PrintedAt { get; set; }
    public bool IsCancelled { get; set; } = false;
    public DateTime? CancelledAt { get; set; }
    public string? CancellationReason { get; set; }
    public string? Notes { get; set; }

    // Navigation properties
    public SalesOrder SalesOrder { get; set; } = null!;
}

File: frontend/types/invoice.ts
export interface Invoice {
  id: number;
  invoiceNumber: string;
  salesOrderId: number;
  salesOrderNumber: string;
  clientId: number;
  clientBusinessName: string;
  clientCuit: string;
  clientTaxCondition: string;
  invoiceDate: string;
  paymentTermId: number | null;
  paymentTermName: string | null;
  usdExchangeRate: number | null;
  specialDiscountPercent: number;
  netAmount: number;
  taxAmount: number;
  totalAmount: number;
  isPrinted: boolean;
  printedAt: string | null;
  isCancelled: boolean;
  cancelledAt: string | null;
  cancellationReason: string | null;
  isCreditNote: boolean;
  isQuotation: boolean;
  notes: string | null;
  createdAt: string;
  updatedAt: string;
  items: InvoiceItem[];
}

export interface InvoiceItem {
  id: number;
  salesOrderItemId: number | null;
  articleId: number;
  articleCode: string;
  articleDescription: string;
  quantity: number;
  unitPriceUsd: number;
  unitPriceArs: number;
  discountPercent: number;
  lineTotal: number;
  createdAt: string;
}

export interface InvoiceListDto {
  id: number;
  invoiceNumber: string;
  invoiceDate: string;
  clientBusinessName: string;
  salesOrderId: number;
  salesOrderNumber: string;
  totalAmount: number;
  isPrinted: boolean;
  isCancelled: boolean;
}

export interface CreateInvoiceRequest {
  salesOrderId: number;
  invoiceDate: string;
  paymentTermId: number | null;
  usdExchangeRate: number | null;
  specialDiscountPercent: number;
  notes: string | null;
}

export interface UpdateInvoiceRequest {
  id: number;
  invoiceDate: string;
  paymentTermId: number | null;
  usdExchangeRate: number | null;
  specialDiscountPercent: number;
  notes: string | null;
}

export interface CancelInvoiceRequest {
  cancellationReason: string;
}
